tests/Doctrine/Tests/ORM/Functional/ParserResultSerializationTest.php

```diff
@@ -61,6 +64,31 @@ public static function provideSerializedSingleSelectResults(): Generator
         yield '2.15.0' => [rtrim(file_get_contents(__DIR__ . '/ParserResults/single_select_2_15_0.txt'), "\n")];
     }
 
+    public function testSymfony44ProvidedData(): void
+    {
+        $sqlExecutor      = $this->createMock(SingleSelectExecutor::class);
+        $resultSetMapping = $this->createMock(ResultSetMapping::class);
+
+        $tempFile = 'symfony_44-temp.php';
+
+        $parserResult = new ParserResult();
+        $parserResult->setSqlExecutor($sqlExecutor);
+        $parserResult->setResultSetMapping($resultSetMapping);
+        $parserResult->addParameterMapping('name', 0);
+
+        $exported = VarExporter::export($parserResult);
+        $data     = file_put_contents($tempFile, '<?php return ' . $exported . ';');
+
+        $unserialized = require $tempFile;
```

Comment:
This is a complicated way to perform an `eval()`, isn't it?

Reply:
What we maybe could do is dump the file once, commit it and then use the output to validate. But then we still need a way to import the file which is the part your question is about. So maybe this is still not the way. 🤔

Reply:
You don't need a file. PHP has `eval()`. Use that.

Reply:
I have pushed a update to use `eval` instead.